### PR TITLE
[ESIMD] Don't force-inline VCStackCall functions.

### PR DESCRIPTION
**Pull request by kbobrovs**
Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp [hunk 1]
```diff
@@ -1650,10 +1650,11 @@ size_t SYCLLowerESIMDPass::runOnFunction(Function &F,
   // There is a current limitation of GPU vector backend that requires kernel
   // functions to be inlined into the kernel itself. To overcome this
   // limitation, mark every function called from ESIMD kernel with
-  // 'alwaysinline' attribute.
+  // 'alwaysinline' attribute, except few cases.
```
Comment (v-klochkov) [inline llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:1653]: Can  you please add one statement explaining why VCFunctionMD::VCStackCall should not be inlined?
Description and comment say nothing about reasoning, which may be permanent, or may become obsolete some day, it is unclear.

Comment (kbobrovs) [reply llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp]: ok, added comments for all the cases where we don't add `alwaysinline`

Comment (v-klochkov) [reply llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp]: Great! Thank you!

**Comment by kbobrovs** [comment]
The failures are unrelated to the patch:
```
  SYCL :: GroupAlgorithm/SYCL2020/exclusive_scan.cpp
  SYCL :: GroupAlgorithm/SYCL2020/inclusive_scan.cpp
  SYCL :: GroupAlgorithm/SYCL2020/reduce.cpp
  SYCL :: GroupAlgorithm/exclusive_scan.cpp
  SYCL :: GroupAlgorithm/inclusive_scan.cpp
  SYCL :: GroupAlgorithm/reduce.cpp
```

**Pull request merged**